src/lib_json/json_reader.cpp

```diff
@@ -1378,30 +1378,39 @@ bool OurReader::readCppStyleComment() {
   }
   return true;
 }
-
-bool OurReader::readNumber(bool checkInf) {
+// if checkNeg is true, read negative number
+bool OurReader::readNumber(bool checkNeg) {
   const char* p = current_;
-  if (checkInf && p != end_ && *p == 'I') {
-    current_ = ++p;
-    return false;
+  char c = '0'; // stopgap for already consumed character  
+  if(checkNeg){
+    if (checkNeg && p != end_ && *p == 'I') {
+      current_ = ++p;
+      return false;
+    }
+    else
+    {
+      current_ = ++p;                // skip '-'
+      OurReader::readNumber(false);  // read the back part of the '-'
+    }
   }
-  char c = '0'; // stopgap for already consumed character
```

Comment:
Why move the `char c` declaration?  We don't use it in the checkNeg block.

Reply:
I move it outside of the checkNeg block.